### PR TITLE
Add pos_file prefix support

### PR DESCRIPTION
**Pull request by alantang888**
Let you able to run multiple fluentd on same node.

### CODE REVIEW

File: templates/conf/kubernetes/cluster-autoscaler.conf.erb [hunk 1]
```diff
@@ -3,9 +3,9 @@
   @id in_tail_cluster_autoscaler
   multiline_flush_interval 5s
   path /var/log/cluster-autoscaler.log
-  pos_file /var/log/fluentd-cluster-autoscaler.log.pos
+  pos_file "#{'/var/log/' + ENV.fetch('FLUENT_POS_PREFIX', '') + 'fluentd-cluster-autoscaler.log.pos'}"
```
Comment (cosmo0920) [inline templates/conf/kubernetes/cluster-autoscaler.conf.erb:6]: How about using File.join to simplify to represent what you wanted to do like as?

```
pos_file "#{File.join('/var/log/', ENV.fetch('FLUENT_POS_PREFIX', ''), 'fluentd-cluster-autoscaler.log.pos')}"
```

Just joining as String instances is dangerous. Safely joining method should be needed to use here.

**Review by cosmo0920 — CHANGES_REQUESTED** [review]

File: templates/conf/kubernetes/cluster-autoscaler.conf.erb (hunk 1 above)
Comment (alantang888) [reply templates/conf/kubernetes/cluster-autoscaler.conf.erb:6]: I just tested a bit. Using File.join will create an addition directory. Not sure is that intention to have addition directory.
If that is the intention I can change to File.join.

---
And I just tested with `../..` with File.join. Look like it work same as string concat on this case. 

I set `FLUENT_POS_PREFIX` == `PREFIX` and `DFLUENT_POS_PREFIX` == `../../PREFIX`.

```
irb(main):001:0> File.join('/var/log/', ENV.fetch('FLUENT_POS_PREFIX', ''), 'fluentd-cluster-autoscaler.log.pos')
=> "/var/log/PREFIX/fluentd-cluster-autoscaler.log.pos"
irb(main):002:0> File.join('/var/log/', ENV.fetch('FLUENT_POS_PREFIX_NOT_EXIST', ''), 'fluentd-cluster-autoscaler.log.pos')
=> "/var/log/fluentd-cluster-autoscaler.log.pos"
irb(main):003:0> File.join('/var/log/', ENV.fetch('DFLUENT_POS_PREFIX', ''), 'fluentd-cluster-autoscaler.log.pos')
=> "/var/log/../../PREFIX/fluentd-cluster-autoscaler.log.pos"
```

Comment (alantang888) [reply templates/conf/kubernetes/cluster-autoscaler.conf.erb:6]: One more question. If added directory. Will fluentd create that directory?? Or will have exception??

Comment (cosmo0920) [reply templates/conf/kubernetes/cluster-autoscaler.conf.erb:6]: From the current Fluentd implementation, Fluentd will create directory if it doesn't exist:
https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/in_tail.rb#L239
This change will be safe.

Comment (alantang888) [reply templates/conf/kubernetes/cluster-autoscaler.conf.erb]: That's great. I modified to use File.join. And I changed the env name to `FLUENT_POS_EXTRA_DIR`. That should more fit for this behaviour.

**Review by cosmo0920 — APPROVED** [review]

**Comment by cosmo0920** [comment]
Could you all take a look thisPR, @ashie @kenhys?
If you all are OK, merge and publish the minor bumping version images, please?

**Comment by ashie** [comment]
Thanks!

**Pull request merged**